sdk/kiosk/src/client/kiosk-client.ts

```diff
@@ -0,0 +1,485 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiClient } from '@mysten/sui.js/src/client';
+import { fetchKiosk, getOwnedKiosks } from '../query/kiosk';
+import {
+	type FetchKioskOptions,
+	KioskData,
+	KioskOwnerCap,
+	ObjectArgument,
+	OwnedKiosks,
+	Network,
+	type KioskClientOptions,
+} from '../types';
+import { TransactionArgument, TransactionBlock } from '@mysten/sui.js/transactions';
+import * as kioskTx from '../tx/kiosk';
+import { TransferPolicyRule, personalKioskAddress, rules } from '../constants';
+import { queryTransferPolicy } from '../query/transfer-policy';
+import { convertToPersonalTx } from '../tx/personal-kiosk';
+import { confirmRequest } from '../tx/transfer-policy';
+import { objArg } from '../utils';
+
+export type PurchaseOptions = {
+	extraArgs?: Record<string, any>;
+};
+
+/**
+ * A Client that allows you to interact with kiosk.
+ * Offers utilities to query kiosk, craft transactions to edit your own kiosk,
+ * purchase, manage transfer policies, create new kiosks etc.
+ */
+export class KioskClient {
+	client: SuiClient;
+	network: Network;
+	#rules: TransferPolicyRule[];
+	selectedCap?: KioskOwnerCap;
+
+	constructor(options: KioskClientOptions) {
+		this.client = options.client;
+		this.network = options.network;
```

Comment:
Could you add a function in SuiClient to expose the `network` such that users don't need to specify the network twice?

cc @Jordan-Mysten the network is needed explicitly because the kiosk package ids are different across networks

Reply:
We don’t really have an easy way to determine the network because there’s no consistent identifier for it in the new client setup. I think the closest you could get would be to query the chain ID and use that.